tools/src/main/java/org/apache/kafka/tools/CommandOptions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+
+/**
+ * Base class for command line tools options
+ */
+public abstract class CommandOptions {
+
+    public ArgumentParser parser;
+    protected Namespace ns;
```

Comment:
If I understood what you mean, it means that from TopicCommandOptions I have to call the CommandOptions constructor after creating the namespace using the parser, so it means that the call to super constructor goes at the end of the TopicCommandOptions constructor. Today it's on top because the CommandOptions ctor creates the parser needed by the rest of the TopciCommandOptions ctor.